packages/orchestrator-ui-components/src/rtk/endpoints/streamMessages.ts

```diff
@@ -0,0 +1,115 @@
+import { debounce } from 'lodash';
+
+import { addToastMessage } from '@/rtk/slices/toastMessages';
+import type { RootState } from '@/rtk/store';
+import { ToastTypes } from '@/types';
+import { getToastMessage } from '@/utils/getToastMessage';
+
+import { CacheTags, orchestratorApi } from '../api';
+
+const getWebSocket = (url: string) => {
+    // TODO: Implement authentication taking this into account: https://stackoverflow.com/a/77060459o
+    return new WebSocket(url);
+};
+
+const PING_INTERVAL = 5000;
+const DEBOUNCE_CLOSE_INTERVAL = 10000;
+
+// From https://redux-toolkit.js.org/rtk-query/usage/customizing-queries#streaming-data-with-no-initial-request
+const streamMessagesApi = orchestratorApi.injectEndpoints({
+    endpoints: (build) => ({
+        streamMessages: build.query<boolean, void>({
```

Comment:
This is quite some block of code, that is also hard to follow to be honest. I would like to suggest to see if there are ways to abstract away some things.

Reply:
I've added more comments with explanation. It's hard and doesn't make it a lot clearer to abstract code away I think.